File: src/main/java/service/DatasCreator.java
package service;

import java.util.List;

import dto.db.ColumnDefinitionDto;

public interface DatasCreator<T> {
	List<T> create(ColumnDefinitionDto def, int rowSize) throws Exception;
	List<T> create(ColumnDefinitionDto def, int rowSize, int startSeed) throws Exception;

	void setSeed(int colIndex);
}

File: src/main/java/tdc/service/DatasCreator.kt
package tdc.service

import tdc.db.ColumnDefinitionDto

interface DatasCreator<T> {
    @Throws(Exception::class)
    fun create(def: ColumnDefinitionDto?, rowSize: Int): List<T>?

    @Throws(Exception::class)
    fun create(def: ColumnDefinitionDto?, rowSize: Int, startSeed: Int): List<T>
    fun setSeed(colIndex: Int)
}